src/Migrator.php

```diff
@@ -226,4 +245,102 @@ protected function migrationTable(string $database = null): Table
     {
         return $this->dbal->database($database)->table($this->config->getTable());
     }
+
+    protected function checkMigrationTableStructure(Database $db): bool
+    {
+        $table = $db->table($this->config->getTable());
+
+        foreach (self::MIGRATION_TABLE_FIELDS_LIST as $field) {
+            if (!$table->hasColumn($field)) {
+                return false;
+            }
+        }
+
+        if (!$table->hasIndex(['migration', 'created_at'])) {
+            return false;
+        }
+
+        return true;
+    }
+
+    /**
+     * Fetch migration information from database
+     *
+     * @param MigrationInterface $migration
+     *
+     * @return array|null
+     */
+    protected function fetchMigrationData(MigrationInterface $migration): ?array
+    {
+        $migrationData = $this->migrationTable($migration->getDatabase())
+            ->select('id', 'time_executed', 'created_at')
+            ->where(
+                [
+                    'migration' => $migration->getState()->getName(),
+                    'created_at' => $this->getMigrationCreatedAtForDb($migration)->format(self::DB_DATE_FORMAT),
+                ]
+            )
+            ->run()
+            ->fetch();
+
+        return is_array($migrationData) ? $migrationData : [];
+    }
+
+    protected function restoreMigrationData(): void
```

Comment:
To cover it by tests we should make some kind of repository for mocking but I suppose it is not required for this issue.

Reply:
Nope.